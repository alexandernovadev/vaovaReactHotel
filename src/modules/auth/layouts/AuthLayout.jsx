import React from "react";
import { Grid, Typography } from "@mui/material";
import { FlagTranslate } from "../../../UI/FlagTranslate";

// interface Props{
//   children: JSX.Element | JSX.Element[];
//   title:string
// }

export const AuthLayout = ({ children, title = "" }) => {
  return (
    <Grid
      className="animate__animated animate__fadeIn"
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justifyContent="center"
      sx={{ 
        minHeight: "100vh",
        backgroundSize:"100% 100%",
        backgroundImage:"url('https://vaovatravel.com/wp-content/uploads/2022/06/YPO-Colombia-2022-419-1-1-scaled.jpg')",
        padding: 4 
      }}
    >
      <Grid
        item
        className="box-shadow animate__animated animate__fadeIn animate__faster"
        xs={3}
        sx={{
          width: { sm: 450 },
          backgroundColor: "primary.light",
          padding: 3,
          borderRadius: 2,
        }}
      >
        <Typography variant="h5" sx={{ mb: 1 }}>
          {title}
        </Typography>
    
        {children}
        <FlagTranslate />
      </Grid>
    </Grid>
  );
};
