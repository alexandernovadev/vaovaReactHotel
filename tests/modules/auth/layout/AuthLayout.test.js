import { render, screen } from "@testing-library/react";
import { AuthLayout } from "../../../../src/modules/auth/layouts/AuthLayout";


describe("Test in AuthLayout", () => {
  test("should to mathc with snapshot", () => {

    const Component = () => <>Im a children :D</>;
    const title = "Testing";


    const { container } = render(
      <AuthLayout title={title} children={<Component />} />
    );
    expect(container).toMatchSnapshot();

  });
});
