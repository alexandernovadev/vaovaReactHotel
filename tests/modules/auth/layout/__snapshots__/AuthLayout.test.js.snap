// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AuthLayout should to mathc with snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-zqh5bb-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 box-shadow animate__animated animate__fadeIn animate__faster css-8ez7fo-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Testing
      </h5>
      Im a children :D
      <button
        class="css-crr5sf"
      >
        en
      </button>
      <b>
        |
      </b>
      <button
        class="css-crr5sf"
        style="font-weight: bold;"
      >
        es
      </button>
    </div>
  </div>
</div>
`;
